python/ray/autoscaler/_private/aws/config.py

```diff
@@ -107,89 +100,105 @@ def log_to_cli(config):
     cli_logger.doassert(provider_name is not None,
                         "Could not find a pretty name for the AWS provider.")
 
+    head_node_type = config["head_node_type"]
+    head_node_config = config["available_node_types"][head_node_type][
+        "node_config"]
+
     with cli_logger.group("{} config", provider_name):
 
         def same_everywhere(key):
-            return config["head_node"][key] == config["worker_nodes"][key]
+            prev_node_config = None
+            for node_type in config["available_node_types"].values():
+                node_config = node_type["node_config"]
+                if prev_node_config is not None:
+                    curr_value = node_config.get(key)
+                    prev_value = prev_node_config.get(key)
+                    if prev_value != curr_value:
+                        return False
+                prev_node_config = node_config
+            return True
 
         def print_info(resource_string,
                        key,
-                       head_src_key,
-                       workers_src_key,
+                       src_key,
                        allowed_tags=None,
                        list_value=False):
             if allowed_tags is None:
                 allowed_tags = ["default"]
 
-            head_tags = {}
-            workers_tags = {}
-
-            if _log_info[head_src_key] in allowed_tags:
-                head_tags[_log_info[head_src_key]] = True
-            if _log_info[workers_src_key] in allowed_tags:
-                workers_tags[_log_info[workers_src_key]] = True
+            node_tags = {}
+            for node_type_key in config["available_node_types"]:
+                node_tags[node_type_key] = {}
+                tag = _log_info[src_key][node_type_key]
+                if tag in allowed_tags:
+                    node_tags[node_type_key][tag] = True
```

Comment:
this will also implicitly include the head node config.

Reply:
Yup, that's fine in this context